File: DSAlgorithms/Leetcode/c++/sliding_window/1838-freq_most_frequent_element.cpp
class Solution {
public:
    int maxFrequency(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int left = 0;
        int right = 0;
        int result = 0;
        long long total = 0;
        while(right < nums.size())
        {
            total+= nums[right];
            while(((long)(right - left + 1) * nums[right]) > total + k)
            {
                total -= nums[left];
                left++;
            }
            result = std::max(result, right - left + 1);
            right++;
        }
       return result;
    }
};

File: DSAlgorithms/Leetcode/ts/src/sliding_window/1838_freq_most_frequent_element.ts
const maxFrequency = (nums: number[], k: number): number =>{
    nums.sort((a, b) => a - b);
    let left = 0;
    let right = 0;
    let result = 0;
    let total = 0;
    while(right < nums.length)
    {
        total += nums[right];
        while(nums[right] * (right - left + 1) > total + k)
        {
            total -= nums[left];
            left++;
        }
        result = Math.max(result, right - left + 1);
        right++;
    }
    return result;
};